src/HieDb/Compat.hs

```diff
@@ -216,3 +217,8 @@ fieldNameSpace_maybe _ = Nothing
 fieldName :: FastString -> NameSpace
 fieldName _ = varName
 #endif
+
+#if __GLASGOW_HASKELL__ < 902
+instance Show ModuleName where
```

Comment:
I don't think we want to expose this orphan, it might conflict with other instances in ghcide etc.

Reply:
This instance used to be defined in HLS itself before it was removed as part of this "drop support for ghc 9.0" PR:
https://github.com/haskell/haskell-language-server/pull/3875/files#diff-380432f16b40730b2d63e7ec74a15219c0efa602f03c884837c9e27cf0cc655bL126-L129

Do you have ideas about how the show instance could be defined for ghc 9.0 in hiedb?

Is it even worth it resolving that TODO or would it be better to just remove it from hls?

A third option that comes to mind: since hls dropped ghc 9.0 support perhaps we could drop support for it in hiedb as well which would make this workaround unnecessary.